src/python_testing/TC_IDM_2_2.py

```diff
@@ -0,0 +1,596 @@
+        # Verify that the DUT does not send the attribute value from cluster A.
+
+        self.print_step(
+            27, "Send the Read Request Message to read any supported attribute/wildcard on a particular cluster say A with the DataVersionFilter Field not set")
+        read_request_27_1_1 = await self.default_controller.ReadAttribute(self.dut_node_id, [Clusters.Objects.Descriptor.Attributes.ServerList])
+
+        data_version_1 = read_request_27_1_1[0][Clusters.Descriptor][Clusters.Attribute.DataVersion]
+        read_request_27_2_1 = await self.default_controller.ReadAttribute(self.dut_node_id, [(0, Clusters.BasicInformation.Attributes.NodeLabel)])
+        data_version_2 = read_request_27_2_1[0][Clusters.BasicInformation][Clusters.Attribute.DataVersion]
+
+        asserts.assert_not_equal(data_version_1, data_version_2)
+
+        # Step 28
+
+        # TH sends a Read Request Message to the DUT to read something(Attribute) which is larger than 1 MTU(1280 bytes) and per spec can be chunked +
+
+        # Verify on the TH that the DUT sends a chunked data message with the SuppressResponse field set to False for all the messages except the last one.
+        # Verify the last chunked message sent has the SuppressResponse field set to True.
+
+        self.print_step(
+            28, "Send the Read Request Message to read something(Attribute) which is larger than 1 MTU(1280 bytes) and per spec can be chunked +")
+        read_request = await self.default_controller.ReadAttribute(self.dut_node_id, "*")
+
+        # Step 29
+
+        # TH sends a Read Request Message to the DUT to read a non global attribute from all clusters at that Endpoint
+        # AttributePath = [[Endpoint = Specific Endpoint, Attribute = Specific Non Global Attribute]] +
+
+        # On the TH verify that the DUT sends an error message and not the value of the attribute.
+        self.print_step(29, "Send the Read Request Message to the DUT to read a non global attribute from all clusters at that Endpoint")
+        read_request = await self.default_controller.ReadAttribute(self.dut_node_id, [Clusters.Objects.Descriptor.Attributes.ServerList])
+
+        # Step 30
+
+        # TH sends a Read Request Message to the DUT to read a non global attribute from all clusters at all Endpoints
+        # AttributePath = [[Attribute = Specific Non Global Attribute]] +
+
+        # On the TH verify that the DUT sends an error message and not the value of the attribute.
+        self.print_step(30, "Send the Read Request Message to the DUT to read a non global attribute from all clusters at all Endpoints")
+        # Hardcode this for now
+        non_global_cluster = Clusters.Objects.LaundryWasherControls
+        read_request = await self.default_controller.ReadAttribute(self.dut_node_id, [(0, non_global_cluster)])
+        asserts.assert_equal(read_request, {})
+
+        # Step 31
+
+        # TH should have access to only a single cluster at one Endpoint1.
```

Comment:
You'll need to set up the ACL to make this happen. And then set it back to the original value after.

Reply:
yes.

Reply:
Added new checks for this